obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+    snr = 999
+    for lineno, line in enumerate(f):
+        if 'headerend' in locals():
+            if len(line.rstrip('\n').rstrip('\r')) in [80, 79] and \
+               (line[79] == ' ' or line[79] == '4' or line[79] == '\n'):
+                pickline += [line]
+        elif line[79] == '7':
+            header = line
+            headerend = lineno
+    amplitude_index = 0
+    for pick_index, line in enumerate(pickline):
+        if line[18:28].strip() == '':  # If line is empty miss it
+            continue
+        station = line[1:6].strip()
+        channel = line[6:8].strip()
+        network = 'NA'  # No network information provided in Sfile.
+        weight = line[14]
+        if weight == '_':
+            phase = line[10:17]
+            weight = 0
+            polarity = ''
+        else:
+            phase = line[10:14].strip()
+            polarity = line[16]
+            if weight == ' ':
+                weight = 0
+        if polarity == '':
+            polarity = "undecidable"
+        elif polarity == 'C':
+            polarity = "positive"
+        elif polarity == 'D':
+            polarity = 'negative'
+        else:
+            polarity = "undecidable"
+        try:
+            time = UTCDateTime(evtime.year, evtime.month, evtime.day,
+                               int(line[18:20]), int(line[20:22]),
+                               int(line[23:28].split('.')[0]),
+                               int(line[23:28].split('.')[1]) * 10000)
+        except (ValueError):
+            time = UTCDateTime(evtime.year, evtime.month, evtime.day,
+                               int(line[18:20]), int(line[20:22]), 0, 0)
+            time += 60  # Add 60 seconds on to the time, this copes with s-file
+            # preference to write seconds in 1-60 rather than 0-59 which
+            # datetime objects accept
+        coda = _int_conv(line[28:33])
+        amplitude = _float_conv(line[33:40])
+        peri = _float_conv(line[41:45])
+        azimuth = _float_conv(line[46:51])
+        velocity = _float_conv(line[52:56])
+        if header[57:60] == 'AIN':
+            ain = _float_conv(line[57:60])
+        elif header[57:60] == 'SNR':
+            snr = _float_conv(line[57:60])
+        azimuthres = _int_conv(line[60:63])
+        timeres = _float_conv(line[63:68])
+        finalweight = _int_conv(line[68:70])
+        distance = _float_conv(line[70:75])
+        caz = _int_conv(line[76:79])
+        # Create a new obspy.event.Pick class for this pick
+        _waveform_id = WaveformStreamID(station_code=station,
+                                        channel_code=channel,
+                                        network_code=network)
+        new_event.picks.append(Pick(waveform_id=_waveform_id,
+                                    phase_hint=phase,
+                                    polarity=polarity, time=time))
+        if line[9] == 'I':
+            new_event.picks[pick_index].onset = 'impulsive'
```

Comment:
All of this `new_event.picks[pick_index]` stuff looks awfully redundant. Why not use a convenience `pick` variable?